olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/StreamLayerClient.h

```diff
@@ -162,6 +163,39 @@ class DATASERVICE_READ_API StreamLayerClient final {
    */
   client::CancellableFuture<UnsubscribeResponse> Unsubscribe();
 
+  /**
+   * @brief Downloads message data using data handle.
+   *
+   * Users should use this method to download data only for messages that
+   * includes data handle and the data size greater than 1 MB. Message with data
+   * size less then 1 MB will have the data embedded.
+   *
+   * @param message  The `Message` instance that was retrieved using `Poll`
+   * method.
+   * @param callback The `DataResponseCallback` object that is invoked when the
+   * get data request is completed.
+   *
+   * @return A token that can be used to cancel this request.
+   */
+  client::CancellationToken GetData(const model::Message& message,
+                                    DataResponseCallback callback);
+
+  /**
+   * @brief Downloads message data using data handle.
```

Comment:
```suggestion
   * @brief Downloads message data using a data handle from the given message.
```
Not entirely clear from where comes the mentioned data handle. Please consider adding more details.